librna/rnalib.c

```diff
@@ -1090,11 +1090,9 @@ static double get_mkpf_value(double x) {
     init = true;
   }
 
-  return exp((-1.0 * x / 100.0) / divisor);
-
   int index = x + HALF;
 
-  if (index >= 0 && index < LOOKUP_SIZE) {
+  if (index >= 0 && index < LOOKUP_SIZE && trunc(x) == x) {
```

Comment:
Adding this check (whether the `double x` input parameter has a remainder equal to 0) fixed the bug/problem. I assumed that the `mk_pf` function was only ever called to scale the `int` values coming from the energy functions (`il_energy` etc.). But apparently there are also some `mk_pf` calls with actual `double` values and not just with `int` values converted to `double` values, which is why the function was performing lookups for some input values it didn't actually precalculate.

Reply:
Unfortunately I'm not familiar with these rules so I can't answer to that, sorry. I didn't encounter this problem during any of my local tests, but one Action kept failing and i eventually figured out that this was the reason. If this function is only ever called with `int` values then this is really confusing since all energy values should indeed fit into `int` values, so i don't exactly know in which situations this check actually would return `false`.

Reply:
might point to a general bug, thus it could be worth following it. Can you open and describe this problem as a new issue?

Reply:
I opened an issue addressing this problem.